script/split_tests.py

```diff
@@ -0,0 +1,197 @@
+#!/usr/bin/env python3
+"""Helper script to split test into n buckets."""
+
+from __future__ import annotations
+
+import argparse
+from dataclasses import dataclass, field
+from math import ceil
+import os
+import subprocess
+import sys
+
+
+class Bucket:
+    """Class to hold bucket."""
+
+    def __init__(
+        self,
+    ):
+        """Initialize bucket."""
+        self.tests = 0
+        self._paths = []
+
+    def add(self, part: TestFolder | TestFile):
+        """Add tests to bucket."""
+        self.tests += part.total_tests
+        self._paths.append(part.path)
+
+    def get_paths_line(self) -> str:
+        """Return paths."""
+        return " ".join(self._paths) + "\n"
+
+
+class BucketHolder:
+    """Class to hold buckets."""
+
+    def __init__(self, tests_per_bucket: int, bucket_count: int) -> None:
+        """Initialize bucket holder."""
+        self._tests_per_bucket = tests_per_bucket
+        self._bucket_count = bucket_count
+        self._current_bucket = Bucket()
+        self._buckets: list[Bucket] = [self._current_bucket]
+        self._last_bucket = False
+
+    def split_tests(self, tests: TestFolder | TestFile) -> None:
```

Comment:
I think that the current approach is a bit naive. Let's say we call this function with test files containing 10, 50, 20 tests and we have a `tests_per_bucket` of 50.

We would end up with 3 buckets containing 10, 50 and 20 tests. However, if we were to first see if there is _any_ bucket to add tests to before moving to the next one, we could end up with 2 buckets, one containing 30 tests, the other 50.

Reply:
One feature of the test groups plugins I've used in the past is randomizing the order on purpose to find tests that rely on side effects from other tests. So actually randomizing the paths list is a good idea!

If we store our file with test order as artifact, we could reproduce the test failure locally easily.